dask/dataframe/core.py

```diff
@@ -4440,25 +4536,29 @@ def __setitem__(self, key, value):
         else:
             df = self.assign(**{key: value})
 
-        self.dask = df.dask
-        self._name = df._name
-        self._meta = df._meta
-        self._divisions = df.divisions
+        self._operation = df.operation
 
     def __delitem__(self, key):
         result = self.drop([key], axis=1)
-        self.dask = result.dask
-        self._name = result._name
-        self._meta = result._meta
+        self._operation = result.operation
 
     def __setattr__(self, key, value):
+        if key == "_operation":
+            object.__setattr__(self, key, value)
+
         try:
             columns = object.__getattribute__(self, "_meta").columns
         except AttributeError:
             columns = ()
```

Comment:
Note that this breaks dask_cudf (since its `DataFrame` initializer doesn't use the `operator=` apporoach internally).